src/dev/lupio/LupioBLK.py

```diff
@@ -41,23 +41,5 @@ class LupioBLK(DmaDevice):
     image = Param.DiskImage("Disk image")
     pio_addr = Param.Addr("Device Address")
     latency = Param.Latency('0ns', "DMA Device Latency")
-    platform = Param.Platform(Parent.any,
-                              "Platform this device is part of.")
-    int_id = Param.Int(0xb, "Interrupt ID");
-
-    def generateDeviceTree(self, state):
-
-        if buildEnv['TARGET_ISA'] != 'riscv':
-            return
-        node = self.generateBasicPioDeviceNode(state, "lupio-blk",
-                                               self.pio_addr,
-                                               self.pio_size)
-        platform = self.platform.unproxy(self)
-        pic = platform.pic
-        node.append(
-                FdtPropertyWords("interrupts", [platform.lupio_blk_int_id]))
-        node.append(
-                FdtPropertyWords("interrupt-parent", state.phandle(pic)))
-        node.appendCompatible(["lupio,blk"])
-        yield node
-
+    pic = Param.Plic("PLIC")
```

Comment:
I think this is the right solution for now. However, we should talk about how to generalize the interrupt controller interface so the block device isn't RISC-V specific.

This would require refactoring a bunch of code in gem5, so it isn't important right now.